d2/src/main/java/com/linkedin/d2/balancer/D2ClientConfig.java

```diff
@@ -61,7 +61,8 @@ public class D2ClientConfig
   @Deprecated
   String zkHosts = null;
   public String xdsServer = null;
-  public String hostName = null;
+  public String hostName = HOST_NAME_DEFAULT;
```

Comment:
Is it possible for the factory to explicitly set this to `null`? In which case the check won't work, even though it may have been loaded through the factory. I'm just curious if there are any ways `isLiRawD2Client` will break

Reply:
it's possible to set null, but users are not doing that, and they won't do it out of no where (no instruction tells them to it). Also, the d2JmxPrefix is a container-specific field that has even less chance to be set manually. So this should cover all cases well.